r2-core/src/main/java/com/linkedin/r2/transport/http/client/AsyncPoolImpl.java

```diff
@@ -697,6 +757,23 @@ private void timeoutObjects()
     return toReap;
   }
 
+  private <U> Collection<TimeTrackingCallback<U>> getExpiredWaiters(Queue<TimeTrackingCallback<U>> queue, long timeout)
```

Comment:
The waiters queue is already a class variable and doesn't need to be passed around, correct?

Reply:
Based this method on existing reap. I have refactored that one as well.